store/postgres/src/notification_listener.rs

```diff
@@ -342,16 +350,39 @@ impl JsonNotification {
             _ => Err(anyhow!("JSON notifications must be numbers or objects"))?,
         }
     }
+}
+
+pub struct NotificationSender {
```

Comment:
could this have `Metrics` in the name? The name implies it actually contains a channel sender.

Reply:
I kinda didn't want to make metrics gathering the name of this struct, it's meant for sending stuff through PG; the naming problem comes more from the ambiguity of `Sender` where it's not clear whether it refers to something about sending with `tokio` or about sending through PG `NOTIFY`. I don't have a great idea for an alternative name. Maybe just a comment on what this struct is for might help?

Reply:
It does seem strange for the `self` in the notify function to be something that just holds the metrics, at that point it could just be a free function with the metrics as a normal argument. But if you feel this refactor will be useful as the sender might grow more fields, I've no objection.